scripts/build-glean-metadata.py

```diff
@@ -44,140 +48,190 @@ def etl_snake_case(line: str) -> str:
+                "name": app.app_id,
+                "description": app.app.get("description", app.app["app_description"]),
+                "channel": app.app.get("app_channel", "release"),
+                "deprecated": app.app.get("deprecated", False),
             }
-            for app in apps
         ]
     )
-)
 
-# Write out some metadata for each app (for the app detail page)
-for app in apps:
-    app_name = app.repo_name
+# sort each set of app ids by the following criteria
+# nightly < beta < release < esr
+# non-deprecated < deprecated
+channel_priority = {"nightly": 1, "beta": 2, "release": 3, "esr": 4}
+for app_group in app_groups.values():
+    app_group["app_ids"].sort(key=lambda app_id: channel_priority[app_id["channel"]])
+    app_group["app_ids"].sort(key=lambda app_id: app_id["deprecated"])
+
+# Write out a list of app groups (for the landing page)
+open(os.path.join(OUTPUT_DIRECTORY, "apps.json"), "w").write(json.dumps(list(app_groups.values())))
+
+# Write out some metadata for each app group (for the app detail page)
+for (app_name, app_group) in app_groups.items():
     app_dir = os.path.join(OUTPUT_DIRECTORY, app_name)
-    (app_ping_dir, app_table_dir, app_metrics_dir) = (
-        os.path.join(app_dir, subtype) for subtype in ("pings", "tables", "metrics")
+    (app_id_dir, app_ping_dir, app_table_dir, app_metrics_dir) = (
+        os.path.join(app_dir, subtype) for subtype in ("app_ids", "pings", "tables", "metrics")
     )
-    for directory in (app_ping_dir, app_table_dir, app_metrics_dir):
+    for directory in (app_id_dir, app_ping_dir, app_table_dir, app_metrics_dir):
         os.makedirs(directory, exist_ok=True)
 
-    app_data = dict(app.repo, pings=[], metrics=[])
-
-    app_id = app.app_id
-    app_id_snakecase = stringcase.snakecase(app_id)
+    app_data = dict(app_group, pings=[], metrics=[])
 
-    # metrics data
-    metrics = app.get_metrics()
+    app_metrics = {}
     metric_pings = dict(data=[])
-    for metric in metrics:  # (metric_name, metric_data) in metrics_data.items():
-        # metrics with associated pings
-        metric_pings["data"].append(
-            {
-                "name": metric.identifier,
-                "description": metric.description,
-                "pings": metric.definition["send_in_pings"],
-                "type": metric.definition["type"],
-                "expires": metric.definition["expires"],
-            }
-        )
+    # keep track of which metric and ping identifiers we have seen so far
+    metric_identifiers_seen = set()
+    ping_identifiers_seen = set()
 
-        app_data["metrics"].append(
-            {
-                "name": metric.identifier,
-                "description": metric.description,
-                "type": metric.definition["type"],
-                "expires": metric.definition["expires"],
-            }
+    for app_id in [app["name"] for app in app_group["app_ids"]]:
```

Comment:
There's a fair bit that has changed, actually, since a bunch of application-id-specific information now goes into subdirectories.

This code is definitely do for a re-work, but I think that particular refactoring would be time consuming (from painful experience, the details here are difficult to get right, see the back/forth above) and not be of a much benefit. 

I'd like to revisit this code/approach when we've finished modifying probe-scraper, possibly in the context of sharing more things between this and mozilla-schema-generator. Hope that's ok.